packages/block-library/src/buttons/deprecated.js

```diff
@@ -1,9 +1,79 @@
+/**
+ * External dependencies
+ */
+import classnames from 'classnames';
 /**
  * WordPress dependencies
  */
-import { InnerBlocks } from '@wordpress/block-editor';
+import { InnerBlocks, useBlockProps } from '@wordpress/block-editor';
+
+/**
+ * @param {Object} attributes Block's attributes.
+ */
+const migrateWithLayout = ( attributes ) => {
+	if ( !! attributes.layout ) {
+		return attributes;
+	}
+
+	const { contentJustification, orientation } = attributes;
+
+	const updatedAttributes = {
+		...attributes,
+	};
+
+	if ( contentJustification || orientation ) {
+		Object.assign( updatedAttributes, {
+			layout: {
+				type: 'flex',
+				justifyContent: contentJustification || 'left',
+				orientation: orientation || 'horizontal',
+			},
+		} );
+	}
+
+	return updatedAttributes;
+};
 
 const deprecated = [
+	{
+		attributes: {
+			contentJustification: {
+				type: 'string',
+			},
+			orientation: {
+				type: 'string',
+				default: 'horizontal',
+			},
+		},
+		supports: {
+			anchor: true,
+			align: [ 'wide', 'full' ],
+			__experimentalExposeControlsToChildren: true,
+			spacing: {
+				blockGap: true,
+				margin: [ 'top', 'bottom' ],
+				__experimentalDefaultControls: {
+					blockGap: true,
+				},
+			},
+		},
+		isEligible: ( { layout } ) => ! layout,
```

Comment:
I don't think we should have `isEligible` here. The thing is that when  this function is checked it gets the `attributes` for this deprecation. This ends up having a newly created `buttons` block (using its default layout) and then in `migrate` the `orientation:horizontal` exists and applies the defaults from the migration.

You can check this by adding a different `default` in `buttons` block.json like:
```
"default": {
	"type": "flex",
	"orientation": "vertical",
	"justifyContent": "right"
}
```

Add a `Buttons` block, save and then reload.

Reply:
Oooh, interesting. How does it work in the [Social Links deprecation](https://github.com/WordPress/gutenberg/blob/trunk/packages/block-library/src/social-links/deprecated.js#L86)?

I can't remove `isEligible` altogether or the deprecation won't work, but I'll change the check to explicitly look for the `contentJustification` and `orientation` attributes, as these are the only ones we need to migrate. If they don't exist we can leave the block as is.

Reply:
All feedback has now been addressed in #36192. Thanks for the review!

Reply:
>How does it work in the Social Links deprecation?

Because:
>The specific handling by `className` below is needed because `itemsJustification`
was introduced in https://github.com/WordPress/gutenberg/pull/28980/files and wasn't
declared in block.json.

So I didn't remove any attributes there.